From 59d855c6b370be083cbd64090f4bdf29584ffa3a Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Fri, 20 Jul 2012 11:32:55 +0300
Subject: [PATCH 259/311] confc.h: define sm_waiter_wait()

sm_waiter_wait() is used in recipes 1 and 4. It's definition has been
omitted by mistake.

LogD #496
RB: r/865/diff/5-6/?file=24144#file24144line229
---
 conf/confc.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/conf/confc.h b/conf/confc.h
index 7665ee6..8479f62 100644
--- a/conf/confc.h
+++ b/conf/confc.h
@@ -217,6 +217,14 @@ struct c2_mutex;
  *         c2_clink_fini(&w->w_clink);
  *         c2_confc_ctx_fini(&w->w_ctx);
  * }
+ *
+ * static int sm_waiter_wait(struct sm_waiter *w)
+ * {
+ *         int rc;
+ *         while ((rc = c2_confc_ctx_error()) == EINPROGRESS)
+ *                 c2_chan_wait(&w->w_clink);
+ *         return rc;
+ * }
  * @endcode
  *
  * <!---------------------------------------------------------------->
-- 
1.8.3.2

